projects/storefrontlib/src/shared/services/trigger/services/trigger.service.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  ComponentFactoryResolver,
+  Injectable,
+  ViewContainerRef,
+} from '@angular/core';
+import {
+  TriggerConfig,
+  TriggerInlineMapping,
+  TriggerMapping,
+  TriggerOutletMapping,
+  TriggerUrlMapping,
+  TRIGGER_CALLER,
+} from '../config/trigger-config';
+import { InlineRenderService } from './inline-render-strategy.service';
+import { OutletRenderService } from './outlet-render-strategy.service';
+import { RoutingRenderService } from './routing-render-strategy.service';
+
+@Injectable({ providedIn: 'root' })
+export class TriggerService {
+  // Keep a list of rendered elements
+  protected renderedCallers: TRIGGER_CALLER[] = [];
+
+  constructor(
+    protected triggerConfig: TriggerConfig,
+    protected componentFactoryResolver: ComponentFactoryResolver,
+    protected outletRenderService: OutletRenderService,
+    protected inlineRenderService: InlineRenderService,
+    protected routingRenderService: RoutingRenderService
+  ) {}
+
+  render(
```

Comment:
it would be really great if we can only pass in the trigger type, and the trigger origin. If we can get the vcr (?) from the trigger component, and handle the rest from the configuration, we have a very clean approach.